vtree/diff.js

```diff
@@ -343,16 +348,23 @@ function reorder(aChildren, bChildren) {
 // TODO: We really want to make this more efficient
 // Finding the current index of any key in the array could be
 // faster.
-function moveItem(array, toIndex, itemKey) {
+function indexOfKey(array, itemKey) {
```

Comment:
I'm sure this is possible. React does it in O(N) with keys: http://facebook.github.io/react/docs/reconciliation.html
I'll think about it more.